lib/apiservers/engine/backends/container.go

```diff
@@ -671,7 +671,11 @@ func (c *Container) ContainerRm(name string, config *types.ContainerRmConfig) er
 	} else {
 		state, err := c.containerProxy.State(vc)
 		if err != nil {
-			return err
+			if IsNotFoundError(err) {
+				cache.ContainerCache().DeleteContainer(id)
```

Comment:
That's kinda what I was getting at in my TODO line in eventmonitor.go....and I can take the time and do that now...but this is much better than what we had...